Fix Default value deserialization from specflow.json

epresi opened the pull request:
<!-- If this is your first PR, please have a look at the Contribution Guidelines (https://github.com/SpecFlowOSS/SpecFlow/blob/master/CONTRIBUTING.md) -->

Fix Default value deserialization from specflow.json.

SpecFlow does not correctly set the following default values from specflow.json:
- generator.allowRowTests
- runtime.obsoleteBehavior
- trace.traceSuccessfulSteps
- trace.minTracedDuration

Highlights:
- language.feature default value was "en" in the code, [but docs says default value is: "en-US"](https://docs.specflow.org/projects/specflow/en/latest/Installation/Configuration.html#language)
- bindingCulture.name default value was "en-US" in the code, [but docs says default value is: not specified](https://docs.specflow.org/projects/specflow/en/latest/Installation/Configuration.html#bindingculture)
- runtime.missingOrPendingStepsOutcome default value was "Pending" in the code, [but docs says default value is: Inconclusive](https://docs.specflow.org/projects/specflow/en/latest/Installation/Configuration.html#runtime)

Is the highlights before means that this PR is a breaking change? @SabotageAndi what do you think?

Fixes #2239, fixes #2439

<!--- Describe your changes in detail -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Performance improvement
- [ ] Refactoring (so no functional change)
- [ ] Other (docs, build config, etc)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- This checklist is here for you that you didn't forget anything -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->



- [x] I've added tests for my code. (most of the time mandatory)
- [x] I have added an entry to the changelog. (mandatory)
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

SabotageAndi commented:
Please adjust the docs and mention it in the changelog.

As it will be in 3.9, I have no problems with breaking changes.

epresi commented:
> Please adjust the docs and mention it in the changelog.
> 
> As it will be in 3.9, I have no problems with breaking changes.

I changed the code so now the **default values are the same** as it is mentioned in the docs. (Before it was different)
I think the only noticeable change by the users is the [runtime.missingOrPendingStepsOutcome](https://docs.specflow.org/projects/specflow/en/latest/Installation/Configuration.html#runtime), because currently the default is `Pending`, however now I changed it to `Inconclusive` so it will align with the docs.

epresi commented:
Reverted my change regarding the `runtime.missingOrPendingStepsOutcome`, and just adjusted the docs.
The default value for it is `Pending` since ~10 years now, so the docs was outdated.
